extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -196,6 +170,51 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
         return getterAccessorsContainer;
     }
 
+    private FieldInfo resolveField(IndexView index, String parameterName, DotName className) {
+        FieldInfoSupplier byParamName = new FieldInfoSupplier(c -> c.field(parameterName), className, index);
+
+        // check if we have field matching the name
+        FieldInfo fieldInfo = byParamName.get();
+        if (parameterName.equals("id")) {
+            // this is a special case where we want to go through the fields of the class
+            // and see if any is annotated with any sort of @persistence.Id/@RestLinkId annotation
+            // N.B. as this module does not depend on any other module that could supply this @Id annotation
+            // (like Panache), we need this general lookup
+            // the order of preference for the annotations is @RestLinkId > @persistence.Id > id
+            FieldInfoSupplier byIdAnnotation = new FieldInfoSupplier(
+                    c -> {
+                        FieldInfo persistenceId = null;
+                        for (FieldInfo field : c.fields()) {
```

Comment:
I made small changes, but I rather not remove the loop as I have no idea what order annotations are listed by looking up fields though annotations. If the order differs from the loop it will break for users that have multiple @Id annotations in a class. What do you think?

Reply:
Good question. Though I doubt fields are ordered either. But @Ladicek will surely know.

Reply:
`ClassInfo.fields()` are sorted by name, although that's probably more of an implementation artifact than a guarantee. Since Jandex 2.4, there's `ClassInfo.unsortedFields()` that returns fields in declaration order, if that's what you need.

Reply:
@Ladicek we wanted to know if ClassInfo.annotations(annotation) orders the result the same as when you loop over ClassInfo.fields(). If so I can improve the code without introducing a breaking change. If not I will have to keep it like this. 

@FroMage Is there a way we can have users migrate to the new RestLinkId annotation which is more safe to use?

Reply:
The order of `ClassInfo.annotations()` is pretty much random. It might be declaration order, but no guarantees really.

Reply:
@Ladicek Thanks for the info. I will have to keep the loop as is.